domains.go

```diff
@@ -1,17 +1,18 @@
 package resend
 
 import (
+	"context"
 	"encoding/json"
 	"errors"
 	"net/http"
 )
 
 type DomainsSvc interface {
-	Create(*CreateDomainRequest) (CreateDomainResponse, error)
-	Verify(domainId string) (bool, error)
-	List() (ListDomainsResponse, error)
-	Get(domainId string) (Domain, error)
-	Remove(domainId string) (bool, error)
+	Create(ctx context.Context, params *CreateDomainRequest) (CreateDomainResponse, error)
```

Comment:
an option to avoid introducing breaking changes would be providing two versions of each public method:
`Method(args)` and `MethodWithContext(ctx, args)`. Where `Method` is a syntax sugar for calling `MethodWithContext` with `Context.Background()`

This is a pattern seen in std libraries such as `net/http` and `database/sql`:
```go
func (db *DB) Query(query string, args ...any) (*Rows, error) { 
    return db.QueryContext(context.Background(), query, args...) 
}
```

So, we would have something like this:
```go
type DomainsSvc interface {
	CreateWithContext(ctx context.Context, params *CreateDomainRequest) (CreateDomainResponse, error)
	Create(*CreateDomainRequest) (CreateDomainResponse, error)
	VerifyWithContext(ctx context.Context, domainId string) (bool, error)
	Verify(domainId string) (bool, error)
	ListWithContext(ctx context.Context) (ListDomainsResponse, error)
	List() (ListDomainsResponse, error)
	GetWithContext(ctx context.Context, domainId string) (Domain, error)
	Get(domainId string) (Domain, error)
	RemoveWithContext(ctx context.Context, domainId string) (bool, error)
	Remove(domainId string) (bool, error)
}
```

Reply:
@gabrielcipriano Great idea! Just refactored as you've proposed, utilizing the Method and MethodWithContext pattern. This approach allows for flexibility while maintaining compatibility and avoiding potential breaking changes.

Reply:
great suggestion @gabrielcipriano